src/atdf/field.rs

```diff
@@ -32,7 +32,20 @@ pub fn parse(
                 bitfield_el,
             )
         })?;
-        chip::ValueRestriction::Enumerated(values.clone())
+        let mask_as_int = util::parse_int(mask)?;
+        let filtered_value: std::collections::BTreeMap<_, _> = values
+            .iter()
+            .filter(|(_, ev)| {
+                if (ev.value & mask_as_int) == ev.value {
+                    true
+                } else {
+                    log::warn!("invalid enumerated value {:?} for mask {}", ev, mask_as_int);
```

Comment:
I was thinking about this a bit more:  Warning on every single enumerated value will result in a _lot_ of warnings ... I think we should only warn once per field.  So instead doing the `warn!()` here, I'd move it further down and warn when `values.len() != filtered_values.len()`.